Apps/Src/main.c

```diff
@@ -2,9 +2,270 @@
 
 #include "common.h"
 #include "config.h"
+#include "Tasks.h"
+#include "stm32f4xx.h"
+#include "CANbus.h"
+#include "Contactors.h"
+#include "Display.h"
+#include "Minions.h"
+#include "MotorController.h"
+#include "Pedals.h"
 
 int main(void) {
-    while (1) {
-        
-    }
+    // Disable interrupts
+    __disable_irq();
+
+    // Initialize some fault bitmaps for error checking purposes
+    OSErrLocBitmap = OS_NONE_LOC;
+    FaultBitmap = FAULT_NONE;
+
+    OS_ERR err;
+    OSInit(&err);
+    assertOSError(OS_MAIN_LOC, err);
+
+    // Initialize apps
+    OSTaskCreate(
+        (OS_TCB*)&Init_TCB,
+        (CPU_CHAR*)"Init",
+        (OS_TASK_PTR)Task_Init,
+        (void*)NULL,
+        (OS_PRIO)TASK_INIT_PRIO,
+        (CPU_STK*)Init_Stk,
+        (CPU_STK_SIZE)WATERMARK_STACK_LIMIT/10,
+        (CPU_STK_SIZE)TASK_INIT_STACK_SIZE,
+        (OS_MSG_QTY)0,
+        (OS_TICK)0,
+        (void*)NULL,
+        (OS_OPT)(OS_OPT_TASK_STK_CLR),
+        (OS_ERR*)&err
+    );
+    assertOSError(OS_MAIN_LOC, err);
+
+    // Enable interrupts
+    __enable_irq();
+
+    // Start OS
+    OSStart(&err);
+    assertOSError(OS_MAIN_LOC, err);
+
+    return 0;
```

Comment:
@ClarkPoon Is it okay for main to return or does it need to go into an infinite while loop

Reply:
If main returns, weird things will happen. I think OSStart and assertOSError should stop us from ever reaching the end of main here